cmake/lint.cmake

```diff
@@ -43,30 +43,34 @@ list(REMOVE_ITEM LINT_SRC
     ${DALI_SRC_DIR}/pipeline/operators/transpose/cutt/LRUCache.h
 )
 
-execute_process(
-  COMMAND ${LINT_COMMAND} --linelength=100 --root=include ${LINT_INC}
-  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-  RESULT_VARIABLE LINT_RESULT
-  ERROR_VARIABLE LINT_ERROR
-  OUTPUT_QUIET
-)
-
-if(LINT_RESULT)
-    message(FATAL_ERROR "Lint failed: ${LINT_ERROR}")
-else()
-    message(STATUS "Lint OK (headers)")
-endif()
+add_custom_target(lint)
 
-execute_process(
-  COMMAND ${LINT_COMMAND} --linelength=100 ${LINT_SRC}
-  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-  RESULT_VARIABLE LINT_RESULT
-  ERROR_VARIABLE LINT_ERROR
-  OUTPUT_QUIET
-)
+foreach(LINT_FILE ${LINT_SRC})
+  string(REPLACE "/" "_" LINT_FILENAME ${LINT_FILE})
+  set(LINT_STAMP_PATH "${CMAKE_CURRENT_BINARY_DIR}/lint/lint_${LINT_FILENAME}_stamp")
+  add_custom_command(
+    OUTPUT ${LINT_STAMP_PATH}
+    DEPENDS ${LINT_FILE}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    COMMAND mkdir -p ${CMAKE_CURRENT_BINARY_DIR}/lint &&
+            touch ${LINT_STAMP_PATH} && ${LINT_COMMAND} --linelength=100 ${LINT_FILE}
+    )
+  add_custom_target(${LINT_FILENAME}
+    DEPENDS ${LINT_STAMP_PATH})
+  add_dependencies(lint ${LINT_FILENAME})
+endforeach(LINT_FILE)
```

Comment:
You could wrap this in function that takes the list of files, root and additional arguments to lint (linelenght).
And just call it twice.

Reply:
Done